### PR TITLE
provider: update azure config with latest

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 1]
```diff
@@ -112,26 +132,6 @@ type AzureConfigSpecAzureNodeOSImage struct {
...
-	MasterSubnetCIDR string `json:"masterSubnetCIDR" yaml:"masterSubnetCIDR"`
-	// WorkerSubnetCIDR is the CIDR for the worker subnet,
-	WorkerSubnetCIDR string                                         `json:"workerSubnetCIDR" yaml:"workerSubnetCIDR"`
-	LoadBalancer     AzureConfigSpecAzureVirtualNetworkLoadBalancer `json:"loadBalancer" yaml:"loadBalancer"`
-}
-
-type AzureConfigSpecAzureVirtualNetworkLoadBalancer struct {
-	// EtcdCidr is the CIDR for the etcd load balancer.
-	EtcdCIDR string `json:"etcdCIDR" yaml:"etcdCIDR"`
-}
```
Comment (kopiczko) [inline pkg/apis/provider/v1alpha1/azure_types.go]: We don't use etcd LB anymore.

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 2]
```diff
@@ -59,46 +59,66 @@ type AzureConfigSpec struct {
...
+	API AzureConfigSpecAzureDNSZonesDNSZone `json:"api" yaml:"api"`
 	// Etcd is the DNS Zone for the etcd cluster.
-	Etcd string `json:"etcd" yaml:"etcd"`
+	Etcd AzureConfigSpecAzureDNSZonesDNSZone `json:"etcd" yaml:"etcd"`
 	// Ingress is the DNS Zone for the Ingress resource, used for customer traffic.
-	Ingress string `json:"ingress" yaml:"ingress"`
+	Ingress AzureConfigSpecAzureDNSZonesDNSZone `json:"ingress" yaml:"ingress"`
 }
 
-type AzureConfigSpecAzureKeyVault struct {
```
Comment (kopiczko) [inline pkg/apis/provider/v1alpha1/azure_types.go]: This is generated instead: https://github.com/giantswarm/azure-operator/pull/70
Less is more.

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 3]
```diff
@@ -59,46 +59,66 @@ type AzureConfigSpec struct {
...
 	VirtualNetwork AzureConfigSpecAzureVirtualNetwork `json:"virtualNetwork" yaml:"virtualNetwork"`
-	Masters        []AzureConfigSpecAzureNode         `json:"masters" yaml:"masters"`
-	Workers        []AzureConfigSpecAzureNode         `json:"workers" yaml:"workers"`
-	DNSZones       AzureConfigSpecAzureDNSZones       `json:"dnsZones" yaml:"dnsZones"`
-}
 
-type AzureConfigSpecAzureResourceGroup struct {
-	Location string `json:"location" yaml:"location"`
+	Masters []AzureConfigSpecAzureNode `json:"masters" yaml:"masters"`
+	Workers []AzureConfigSpecAzureNode `json:"workers" yaml:"workers"`
```
Comment (kopiczko) [inline pkg/apis/provider/v1alpha1/azure_types.go]: Left outside alphabetical order. I'd like to extract `nodes` field to extract some common stuff like AzureConfigSpecAzureNodeOSImage.

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 4]
```diff
@@ -59,46 +59,66 @@ type AzureConfigSpec struct {
...
-	KeyVault       AzureConfigSpecAzureKeyVault       `json:"keyVault" yaml:"keyVault"`
-	ResourceGroup  AzureConfigSpecAzureResourceGroup  `json:"resourceGroup" yaml:"resourceGroup"`
-	Storage        AzureConfigSpecAzureStorage        `json:"storage" yaml:"storage"`
+	DNSZones    AzureConfigSpecAzureDNSZones    `json:"dnsZones" yaml:"dnsZones"`
+	HostCluster AzureConfigSpecAzureHostCluster `json:"hostCluster" yaml:"hostCluster"`
+	// Location is the region for the resource group.
+	Location string `json:"location" yaml:"location"`
+	// StorageSKUName is the name of storage capability.
+	// https://docs.microsoft.com/en-us/rest/api/storagerp/StorageAccounts/Create#definitions_skuname
+	StorageSKUName string                             `json:"storageSKUName" yaml:"storageSKUName"`
```
Comment (kopiczko) [inline pkg/apis/provider/v1alpha1/azure_types.go]: This is flattened. It'll be removed anyway when we switch to vm disks.

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 5]
```diff
@@ -59,46 +59,66 @@ type AzureConfigSpec struct {
...
+	// used to determine DNS hosted zone to put NS records in.
+	ResourceGroup string `json:"resourceGroup" yaml:"resourceGroup"`
+}
+
+type AzureConfigSpecAzureVirtualNetwork struct {
+	// CIDR is the CIDR for the Virtual Network.
+	CIDR string `json:"cidr" yaml:"cidr"`
+	// MasterSubnetCIDR is the CIDR for the master subnet,
+	MasterSubnetCIDR string `json:"masterSubnetCIDR" yaml:"masterSubnetCIDR"`
+	// WorkerSubnetCIDR is the CIDR for the worker subnet,
```
Comment (xh3b4sd) [inline pkg/apis/provider/v1alpha1/azure_types.go]: `,` -> `.`

File: pkg/apis/provider/v1alpha1/azure_types.go [hunk 6]
```diff
@@ -59,46 +59,66 @@ type AzureConfigSpec struct {
...
-	DataDiskSizeGB int `json:"dataDiskSizeGB" yaml:"dataDiskSizeGB"`
 	// AdminUsername is the vm administrator username
 	AdminUsername string `json:"adminUsername" yaml:"adminUsername"`
 	//  AdminSSHKeyData is the vm administrator ssh public key
 	AdminSSHKeyData string `json:"adminSSHKeyData" yaml:"adminSSHKeyData"`
+	// DataDiskSizeGB is the vm data disk size in GB
+	DataDiskSizeGB int `json:"dataDiskSizeGB" yaml:"dataDiskSizeGB"`
 	// OSImage is the vm OS image object
 	OSImage AzureConfigSpecAzureNodeOSImage `json:"osImage" yaml:"osImage"`
+	// VMSize is the master vm size (e.g. Standard_A1)
```
Comment (xh3b4sd) [inline pkg/apis/provider/v1alpha1/azure_types.go]: `.`

File: pkg/apis/provider/v1alpha1/azure_types.go (hunk 5 above)
Comment (kopiczko) [reply pkg/apis/provider/v1alpha1/azure_types.go]: Thanks, done.

File: pkg/apis/provider/v1alpha1/azure_types.go (hunk 6 above)
Comment (kopiczko) [reply pkg/apis/provider/v1alpha1/azure_types.go]: Thanks, done.

**Pull request merged**